Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/CommonComponents/LocationDrawingsButton.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { BtnDropdown, GenericController, LoadingScreen, ServerErrorIcon, ToolTip } from '@gpa-gemstone/react-interactive';
+import LocationDrawingsModal from './LocationDrawingsModal';
+import { OpenXDA } from '@gpa-gemstone/application-typings';
+import { CrossMark } from '@gpa-gemstone/gpa-symbols';
+
+interface LocationDrawingsButtonProps {
+    Locations: OpenXDA.Types.Location[];
+}
+
+const LocationDrawingsButton: React.FC<LocationDrawingsButtonProps> = (props) => {
+    const [hover, setHover] = React.useState<'none' | 'drawings'>('none');
+    const [pageState, setPageState] = React.useState<"loading" | "error" | "idle">("idle");
+    const [selectedLocation, setSelectedLocation] = React.useState<OpenXDA.Types.Location>();
+    const [multipleLocations, setMultipleLocations] = React.useState<boolean>(false);
+    const [showDrawingsModal, setShowDrawingsModal] = React.useState<boolean>(false);
+    const [locationsWithErrors, setLocationsWithErrors] = React.useState<Map<number, string[]>>(new Map())
+    const LocationDrawingController = new GenericController(`${homePath}api/LocationDrawing`, "Name", true);
+
+    const isValid = (location: OpenXDA.Types.Location, drawingData) => {
+        let e = [];
+
+        if (!location
+            || (location.Alias == ""
+                && location.Description == ""
+                && location.ID == 0
+                && location.Latitude == null
+                && location.LocationKey == ""
+                && location.Longitude == null
+                && location.Name == ""))
+            e.push('No locations have been set.');
+        else if (drawingData.TotalRecords == 0)
+            e.push('No drawings associated with location.');
+        return e;
+    }
+
+    React.useEffect(() => { // Generates the map of errors for each location
+        if (props.Locations.length > 1) setMultipleLocations(true);
```

Comment:
does `multipleLocations` really need to be a state? `react.useMemo(() => props.Locations.length > 1,[props.Locations]);` seems more appropriate